book/integration-service.md

```diff
@@ -0,0 +1,264 @@
+  - Triggering of Composite PipelineRun
+  - Triggering a Promotion to Staging Environment
+  - Triggering a Release to Production Environment
+
+Is that the case still?
+
+If it is, StoneSoup Build Pipeline was designed in a way that none of its tasks should fail so that the rest of tasks in the graph can execute. The results of some of these tasks, Sanity and Security Testing, are evaluated later on by the Enterprise Contract. 
+
+This same problem will exist for the Test pipeline. Initially we can allow the task to fail which would then cause the PipelineRun to fail. But if we plan to support application teams to define a series of Environment Provisioning Tasks and Test Tasks to execute tests in parallel within the same pipeline. This does not work well. So if an Environment Provisioning Task fails for a temporal issue, or a Test task fails due to a test failure, the other test tasks in the graph would not be able to execute. There are two main areas to this dilemma:
+
+  - How to design the tekton pipeline/task to allow the graph to continue to execute [Issue 1684](https://github.com/tektoncd/pipeline/issues/1684) which has spun off a few TEPS to try to mitigate:
+
+    - The [Finally](https://tekton.dev/vault/pipelines-v0.16.3/pipelines/#adding-finally-to-the-pipeline) clause was introduced
+    - [TEP0040](https://github.com/jerop/community/blob/22ad7f6ae9c558ce66d74c1186d3a528cf846677/teps/0040-ignore-step-errors.md) Task Step OnError (Alpha only)
+    - [TEP0050](https://github.com/jerop/community/blob/22ad7f6ae9c558ce66d74c1186d3a528cf846677/teps/0050-ignore-task-failures.md) Ignore Task Failures (Proposed)
+    - [TEP0056](https://github.com/tektoncd/community/blob/main/teps/0056-pipelines-in-pipelines.md) Pipeline in Pipeline (Proposed)
+    - [TEP0049](https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md) Task Results without Results (Proposed)
+
+  - How to get the results in times of failure. [Issue #3749](https://github.com/tektoncd/pipeline/issues/3749)
+    - [TEP0044](https://github.com/tektoncd/community/blob/main/teps/0044-data-locality-and-pod-overhead-in-pipelines.md) Data Locality and Pod Overhead in Pipelines (proposed)
+
+
+### Proposal
+We should be consistent and continue to design the Test pipelines so that the tasks don’t fail and continue to produce the HACBS_TEST_OUTPUT.
+
+#### Option 1
+leverage the enterprise contract task to run the test validation rego policy against this pipeline.
+
+#### Option 2
+
+If being a part of the Enterprise Contract is NOT possible and not clear. We should still implement the same technology and mechanism we do now with EC. What we can do is extend the IntegrationTestScenario spec to have a testPolicyWhitelist field that a user defines per Scenario. This Policy will actually be a git repo to a data file that can be used by a conftest/opa rego file. 
```

Comment:
Red Hat is trying to replace blacklist/whitelist with blocklist/allowlist instead. Hopefully it's not too late to change the field name?

Reply:
The entire `open questions` section is actually an enhancement proposal and hasn't been implemented as such.
@srivickynesh since a lot of use cases that this proposal addresses has been covered with the other parts of the design already (such as IntegrationTestScenario having contexts and marking them as optional/required), my proposal is to drop this section.

Reply:
Done

Reply:
Thanks @srivickynesh 
@amfred we'll certainly keep in mind to use blocklists/allowlists, thanks for pointing that out.